cdap-api/src/main/java/co/cask/cdap/api/mapreduce/MapReduceContext.java

```diff
@@ -63,7 +64,9 @@
    * Overrides the input configuration of this MapReduce job to use the specific stream.
    *
    * @param stream the input stream.
+   * @deprecated as of 3.4.0. Use {@link MapReduceContext#addInput(Input)} instead.
    */
+  @Deprecated
   void setInput(StreamBatchReadable stream);
```

Comment:
Documented this behavior on the `addInput` methods.
We need this behavior because the `setInput(InputFormatProvider)` method doesn't specify an input name, so we just use the classname as the input name.
So, we don't want users to mix usage of that API with `addInput` methods.